docs/express-vs-core.qmd

```diff
@@ -0,0 +1,262 @@
+
+Meanwhile, the `reactive` module is unchanged between Core and Express, as the two modes share the same reactive foundations.
+
+Finally, notice that Express also imports `input` from `shiny.express`.
+This isn't needed in Core, because the `input` object is passed into the server function as an argument.
+Since Express has no server function, we made it an attribute of `shiny.express`.
+
+### Organization of UI and server code
+
+**Every Shiny Core app file has a UI section, and a server section;** the two are separate and distinct.
+
+The UI section consists of a single (potentially very long and deeply nested) expression, stored as a variable named `app_ui` by convention.
+The object this produces is actually straight up HTML, which is sent to the browser when it first loads the app.
+
+The server section is a function, named `server` by convention, that always takes the arguments `input`, `output`, and `session`.
+This function contains render functions and reactive functions, which are used to update the UI in response to user input.
+
+You can think of the UI section as a template, with `ui.output_xx()` calls as placeholders for outputs, and the server section as the instructions for how to populate those outputs.
+
+```{shinylive-python}
+#| standalone: true
+#| components: [editor, viewer]
+#| layout: vertical
+# Core
+from shiny import ui, render, reactive, App
+from datetime import datetime
+
+app_ui = ui.page_fixed(
+    ui.h1("Title"),
+    ui.output_code("greeting"),
+)
+
+def server(input, output, session):
+    @reactive.Calc
+    def time():
+        reactive.invalidate_later(1)
+        return datetime.now()
+
+    @render.code
+    def greeting():
+        return f"Hello, world!\nIt's currently {time()}."
+
+app = App(app_ui, server)
+```
+
+**In Shiny Express, there isn't this hard distinction** between UI and server.
+
+Instead, everything coexists as top-level code in the app file: input components, layout directives, outputs (including their rendering logic), and reactive functions.
+
+Also, you aren't forced to combine your UI into a single complex object.
+You can use multiple UI objects, and they will be combined together by the framework.
+
+```{shinylive-python}
+#| standalone: true
+#| components: [editor, viewer]
+#| layout: vertical
+# Express
+from shiny import ui, render, reactive, App
+import shiny.express
+from datetime import datetime
+
+ui.h1("Title")
+
+@reactive.Calc
+def time():
+    reactive.invalidate_later(1)
+    return datetime.now()
+
+@render.code
+def greeting():
+    return f"Hello, world!\nIt's currently {time()}."
+```
+
+Again, notice how `greeting` in this app does _not_ have a corresponding call to `output_code("greeting")`.
+This is because in Shiny Express, `render` functions automatically add an output to the page---no need to do it manually.
```

Comment:
```suggestion
This is because in Shiny Express, `render` functions implicitly add an output to the page as a side effect so you don't need the output function.
```